### ISSUE TITLE
Async instance edit screen load

### ISSUE THREAD

**Issue by BoBkiNN** [PrismLauncher/PrismLauncher#3147]
### Role

I often edit instances

### Suggestion

Make instance edit window load async

### Benefit

App will not freeze when edit button is clicked and window will show earlier

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

When i click first time (since app open) on instance edit button to open logs, see mods and etc for instance it takes about 15 seconds for window to open, until its opened main page is freezed and application is not responding

**Comment by TheKodeToad** [comment]
I don't think this is really needed... what IS needed is to stop it from taking 15 seconds - which should be fixed in the next version
Well, it is done by making a http request async. I don't think making everything async is that helpful

**Comment by TauCu** [comment]
My edit windows took around 10-15 seconds to load, after updating from 9.2 to 9.4 it now takes over 30 seconds.

By viewing PrismLauncher's IO in procmon it appears to be walking the directory tree of the saves. The problem is I have a lot of saves, some of them from servers (and hence >10k playerdata files each version)

Only walk this data when you click "worlds"
Ideally it should also be done async, but I'd settle for it not being done until needed.

![Image](https://github.com/user-attachments/assets/e9ecfff8-1764-4b47-9955-f8453e04a821)

**Comment by Vishrut2403** [comment]
#5845 fixes this issue I think or partially?

**Comment by Octol1ttle** [comment]
PRs for async world load:
https://github.com/PrismLauncher/PrismLauncher/pull/3651
https://github.com/PrismLauncher/PrismLauncher/pull/5845

This should bring loading times back to normal
